# Análisis de datos e insights

Este notebook contiene las consultas resultados y el análisis de la integración de datos.

In [ ]:
import psycopg2
import pandas as pd
from sqlalchemy import create_engine
engine = create_engine("postgresql://admin:1234@localhost:5433/metrics")

### Insight 1 - Países más económicos para el turismo

In [16]:
query = """
SELECT 
  pais,
  AVG(costo_bajo_hospedaje + costo_promedio_comida + 
  costo_bajo_transporte + costo_promedio_entretenimiento) 
  AS costo_total
FROM metrics
GROUP BY pais
ORDER BY costo_total ASC
LIMIT 5;
"""
df = pd.read_sql(query, engine)
df


pais  costo_total
0   Namibia        58.00
1    Greece        72.36
2  Tanzania        75.00
3   Etiopía        77.00
4   Iceland        77.76

Los países más económicos para el el turismo son Namibia, Grecia, Tanzania, Etiopía e Islandia.


Se recomienda que estos países promuevan más el turismo hacia sus países debido a que pueden ser destinos populares para los mochileros o para aquellas personas que no desean gastar mucho al vacacionar. Asimismo, podrían ofrecer paquetes integrales que le permitan al turista disfrutar del país en todos los ámbitos. Esto resultaría en un crecimiento económico significativo para el país si se ejecuta de manera adecuada.

### Insight 2 - Países con menor tasa de envejecimiento y entretenimiento más barato

In [17]:
query = """
SELECT 
  pais,
  tasa_de_envejecimiento,
  costo_promedio_entretenimiento
FROM metrics
WHERE tasa_de_envejecimiento < 10
  AND costo_promedio_entretenimiento < 30
ORDER BY tasa_de_envejecimiento ASC, costo_promedio_entretenimiento ASC
LIMIT 10;


"""
df = pd.read_sql(query, engine)
df


pais  tasa_de_envejecimiento  \
0                Serbia                    5.11   
1                  Irán                    5.41   
2              Paraguay                    5.51   
3  República Dominicana                    6.49   
4            San Marino                    6.77   
5              Bulgaria                    7.16   
6       North Macedonia                    7.79   
7                Norway                    7.82   
8              Slovakia                    8.12   
9               Hungary                    8.12   

   costo_promedio_entretenimiento  
0                           29.16  
1                           26.00  
2                           21.00  
3                           29.00  
4                           20.52  
5                           21.60  
6                           22.68  
7                           29.16  
8                           19.44  
9                           19.44

Países como Paraguay, San Marino, Slovakia y Hungary ofrecen costos de entretenimiento menores a $22, y muestran tasas de envejecimiento por debajo de 8%. Esto indica una demografía joven con opciones de entretenimiento económicas lo que es ideal para considerar el impulso de rutas turísticas estudiantiles, actividades recreativas al aire libre o culturales y programas de work and travel.

Se recomienda que los países realicen promociones por medio de redes sociales enfocadas a los jóvenes viajeros, desarrollo de festivales y eventos culturales, realizar alianzas con instituciones educativas para promover el turismo académico y promover un alojamiento económico para los jóvenes.

### Insight 3 - Continentes con mayor promedio de gasto total por país

In [18]:
query = """
SELECT 
  continente,
  AVG(costo_bajo_hospedaje + costo_promedio_comida + 
  costo_bajo_transporte + costo_promedio_entretenimiento) 
  AS gasto_promedio
FROM metrics
GROUP BY continente
ORDER BY gasto_promedio DESC;
"""
df = pd.read_sql(query, engine)
df


continente  gasto_promedio
0    América      153.200000
1       Asia      131.700000
2     Europa      105.769565
3     África       96.950000

En base a los resultados se puede observar que el continente más caro es América con un gasto promedio aproximado de $153.20. Esto sugiere que este continente suele ofrecer experiencias más costosas que posiblemente se realcionen con el turismo urbano.

Mientras que el continente con el gasto promedio mas barato es África, con  un gasto promedio aproximado de $96.95. Esto podría motivar a viajeros con presupuestos menores o a aquellos que buscan la aventura y la naturaleza a considerar África como opción.


Se recomienda que África desarrolle paquetes económicos y facilite las vías de transporte hacia al continente para que mas personas lo tomen en consideración a la hora de realizar planes. Mientras que a América se le recomienda promover sus destinos como lujosos y exclusivos por sus costos.